digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```
@@ -687,17 +687,26 @@ class Digiroad2Api(val roadLinkService: RoadLinkService,
     }
   }
 
-  get("/verificationInfo") {
-    val typeId = params.getOrElse("typeId", halt(BadRequest("Missing mandatory 'typeId' parameter"))).toInt
+  get("/getMunicipalityInfo") {
     params.get("bbox").map { bbox =>
       val boundingRectangle = constructBoundingRectangle(bbox)
       validateBoundingBox(boundingRectangle)
-      verificationService.getMunicipalityInfo(typeId, boundingRectangle)
+      verificationService.getMunicipalityInfo(boundingRectangle)
     } getOrElse {
       BadRequest("Missing mandatory 'bbox' parameter")
     }
   }
 
+  get("/verificationInfo") {
+    val typeId = params.getOrElse("typeId", halt(BadRequest("Missing mandatory 'typeId' parameter"))).toInt
+    val municipalityCode = params.getOrElse("municipality", halt(BadRequest("Missing mandatory 'municipality' parameter"))).toInt
+    val user = userProvider.getCurrentUser()
+    if (user.isAuthorizedToRead(municipalityCode))

```

Instead of returning None it should throw a exception non authorized.